src/track/beatgrid.cpp

```diff
@@ -369,3 +388,34 @@ void BeatGrid::setBpm(double dBpm) {
     locker.unlock();
     emit updated();
 }
+
+void BeatGrid::setSignature(mixxx::TimeSignature sig, double dSample) {
+    Q_UNUSED(dSample);
+    QMutexLocker locker(&m_mutex);
+    if (!isValid())
+        return;
+
+    m_grid.mutable_first_beat()->mutable_signature()->set_beats(sig.getBeats());
+    m_grid.mutable_first_beat()->mutable_signature()->set_note_value(sig.getNoteValue());
+}
+
+mixxx::TimeSignature BeatGrid::getSignature(double dSample) const {
+    Q_UNUSED(dSample);
+    QMutexLocker locker(&m_mutex);
+    if (!isValid())
+        return mixxx::TimeSignature(0,0);
+
+    // If no signature assume 4/4
+    if (!m_grid.first_beat().has_signature()) {
+        return mixxx::TimeSignature(4,4);
+    } else {
+        return mixxx::TimeSignature(
```

Comment:
Does 
return m_grid.first_beat().signature()
work?

Reply:
BeatGrid is going to disappear so I will apply take the comment into account while working on the new Beats class..

Reply:
@daschuer  Sorry, after all that time I have lost the context of you question.Can you please re elaborate?